### PR TITLE
Bump docusaurus to v3.1.0

### PR DESCRIPTION
**Pull request by maxrake**
This change bumps the version of docusaurus from v3.0.1 to [v3.1.0](https://github.com/facebook/docusaurus/releases/tag/v3.1.0) and accounts for the new features found in the release. Namely, the new `onBrokenAnchors` configuration is added and set to `throw` to match the existing related settings. Additionally, a typo in `CONTRIBUTING.md` was fixed.

### CODE REVIEW

**Comment by maxrake** [comment]
Phylum analysis was run locally but still shows as `INCOMPLETE` due to a number of unprocessed packages. Waiting for those to complete is recommended before proceeding with a merge of this PR.

**Comment by maxrake** [comment]
Phylum analysis completed and was a success. This PR is ready to be merged, pending an approval.

**Review by janasheehan — APPROVED** [review]
lgtm

**Pull request merged**
